common/log/class.SingleFileAppender.php

```diff
@@ -136,11 +136,17 @@ protected function initFile()
     {
         if ($this->maxFileSize > 0 && file_exists($this->filename) && filesize($this->filename) >= $this->maxFileSize) {
             // need to reduce the file size
-            $file = file($this->filename);
-            $file = array_splice($file, ceil(count($file) * $this->reduceRatio));
+            $reduceLen= ($this->maxFileSize< (int)ini_get('memory_limit')*1024?$this->maxFileSize:(int)ini_get('memory_limit')*1024)*$this->reduceRatio;
+            $size = filesize($this->filename)-$reduceLen;
+            $log =file_get_contents($this->filename, false, null,  $size, $reduceLen );
```

Comment:
Try to be consistent with spacing. For example always surround `=` with spaces (eg. `$log = file_get_contents(...`)

Reply:
Also please omit the whitespace after the last argument (in this case `$reduceLen`)